aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -158,15 +160,18 @@ def __init__(
             provided http headers
         body: Union[str, bytes, None]
             Optionally set the response body. Note: bytes body will be automatically base64 encoded
-        headers: dict
-            Optionally set specific http headers. Setting "Content-Type" hear would override the `content_type` value.
+        headers: dict[str, List[str]]
+            Optionally set specific http headers. Setting "Content-Type" here would override the `content_type` value.
+        cookies: list[str]
+            Optionally set cookies.
         """
         self.status_code = status_code
         self.body = body
         self.base64_encoded = False
-        self.headers: Dict = headers or {}
+        self.headers: Dict[str, List[str]] = defaultdict(list, **headers) if headers else defaultdict(list)
+        self.cookies = cookies or []
         if content_type:
-            self.headers.setdefault("Content-Type", content_type)
+            self.headers.setdefault("Content-Type", [content_type])
```

Comment:
For my own benefit, what are the chances we give customers a chance to shoot themselves in the foot?

e.g., setting multiple content types

Reply:
On this PR we are exposing headers as `Dict[str, List[str]]`. Besides `Cookies` I'm struggling to think about examples were a webserver sets multiple headers with the same key. Let's discuss this.